pkg/client/portforward/portforward_test.go

```diff
@@ -206,6 +206,102 @@ func (s *fakeUpgradeStream) Headers() http.Header {
 	return http.Header{}
 }
 
+func TestGetListener(t *testing.T) {
+	stopChan := make(chan struct{}, 1)
+	pf, err := New(&client.Request{}, &client.Config{}, []string{"15000", "15000"}, stopChan)
+	if err != nil {
+		t.Fatalf("Cannot create PortForwarder %v: %v", pf, err)
+
+	}
+	upgrader := fakeUpgrader{conn: newFakeUpgradeConnection()}
+	pf.upgrader = &upgrader
+	testCases := []struct {
+		Hostname                string
+		Protocol                string
+		Ports                   ForwardedPort
+		ShouldRaiseError        bool
+		ExpectedListenerAddress string
+		ExpectedListenerPort    string
+	}{
+		{
+			Hostname:                "localhost",
+			Protocol:                "tcp4",
+			Ports:                   ForwardedPort{15001, 15001},
+			ShouldRaiseError:        false,
+			ExpectedListenerAddress: "127.0.0.1",
+			ExpectedListenerPort:    "15001",
+		},
+		{
+			Hostname:                "127.0.0.1",
+			Protocol:                "tcp4",
+			Ports:                   ForwardedPort{15002, 15002},
+			ShouldRaiseError:        false,
+			ExpectedListenerAddress: "127.0.0.1",
+			ExpectedListenerPort:    "15002",
+		},
+		{
+			Hostname:                "[::1]",
+			Protocol:                "tcp6",
+			Ports:                   ForwardedPort{15003, 15003},
+			ShouldRaiseError:        false,
+			ExpectedListenerAddress: "::1",
+			ExpectedListenerPort:    "15003",
+		},
+		{
+			Hostname:                "localhost",
+			Protocol:                "tcp6",
+			Ports:                   ForwardedPort{15004, 15004},
+			ShouldRaiseError:        false,
+			ExpectedListenerAddress: "::1",
+			ExpectedListenerPort:    "15004",
+		},
+		{
+			Hostname:         "[::1]",
+			Protocol:         "tcp4",
+			Ports:            ForwardedPort{16001, 16001},
+			ShouldRaiseError: true,
+		},
+
+		{
+			Hostname:         "www.google.com",
+			Protocol:         "tcp4",
+			Ports:            ForwardedPort{16002, 16002},
+			ShouldRaiseError: true,
+		},
+		{
+			Hostname:         "1.1.1.1",
```

Comment:
It's to make sure that we can get the port from "localhost:$port", where $port may be 0, in which case the listener's Addr will contain an arbitrary ephemeral port. Please remove tests for anything other than localhost an ::1.